pkg/plugins/crypto/api.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package crypto
+
+import "github.com/google/tink/go/keyset"
+
+// Provider interface for Crypto ctx
+type Provider interface {
+}
+
+// Creator method to create new pluggable Crypto service
+type Creator func(prov Provider) (AriesCrypto, error)
+
+// AriesCrypto is an interface defining a pluggable Crypto operations provider to Aries agents.
+// It's only responsible for pure crypto operations and nothing more. Envelope building and keys management are managed
+// in other interfaces
+type AriesCrypto interface {
+	// Encrypt will encrypt msg and aad using a matching AEAD primitive in kh
+	// returns:
+	// 		cipherText in []byte
+	//		nonce in []byte
+	//		error in case of errors during encryption
+	Encrypt(msg, aad []byte, kh *keyset.Handle) ([]byte, []byte, error)
```

Comment:
Is `keyset.Handle` extendable by 3rd party crypto implementations?

Reply:
from the doc comment:
```
// Handle provides access to a Keyset protobuf, to limit the exposure of actual protocol
// buffers that hold sensitive key material.
```
Tink serializes all keys as protobufs, the Handle has the primitive info (cipher suite) needed for each of the enabled keys in the keyset.

Reply:
I was thinking to use `*interface{}` type but `keyset.Handle` is a struct

Reply:
I am not sure I understand the problem. You can wrap a struct and meet an interface in the wrapper.